include/git2/diff.h

```diff
@@ -347,6 +347,21 @@ typedef int (*git_diff_notify_cb)(
 	void *payload);
 
 /**
+ * Diff progress callback.
+ *
+ * Called before each file comparison.
+ *
+ * - 'old_path' is the path to the old file or NULL
+ * - 'new_path' is the path to the new file or NULL
+ *
+ * Returning a non-zero value will abort the diff.
+ */
```

Comment:
The documentation has a specific format so it can be parsed by tools. The parameters have `@param`, return has `@return` etc.